R/derive_summary_records.R

```diff
@@ -149,14 +204,47 @@ derive_summary_records <- function(dataset,
     set_values_to <- exprs(!!analysis_var := {{ summary_fun }}(!!analysis_var), !!!set_values_to)
   }
 
-  # Summarise the analysis value and bind to the original dataset
-  bind_rows(
-    dataset,
-    get_summary_records(
-      dataset,
-      by_vars = by_vars,
-      filter = !!filter,
-      set_values_to = set_values_to
+  if (is.null(dataset_add)) {
+    dataset_add <- dataset
+  }
+
+  summary_records <- dataset_add %>%
+    group_by(!!!by_vars) %>%
+    filter_if(filter) %>%
+    summarise(!!!set_values_to) %>%
+    ungroup()
+
+  if (!is.null(missing_values)) {
+    update_missings <- map2(
+      syms(names(missing_values)),
+      missing_values,
+      ~ expr(if_else(is.na(!!.x), !!.y, !!.x))
     )
+    names(update_missings) <- names(missing_values)
+    summary_records <- summary_records %>%
+      mutate(!!!update_missings)
+  }
```

Comment:
I think `missing_values` should be applied to `new_ref_obs` below.

Reply:
The `missing_values` argument is intended to define the values for by groups which are not in `dataset_add`. For example, if you derive the total dose, you could use `missing_values = exprs(AVAL = 0)` to set the total dose to zero for patients without exposure records.

Reply:
Got it, that makes much more sense